osm2streets/src/operations/collapse_short_road.rs

```diff
@@ -51,34 +45,15 @@ impl StreetNetwork {
             let mut trim_roads_for_merging = BTreeMap::new();
             for i in [keep_i, destroy_i] {
                 for road in self.roads_per_intersection(i) {
-                    // If we keep this in there, it might accidentally overwrite the
-                    // trim_roads_for_merging key for a surviving road!
-                    if road.id == short_r {
-                        continue;
-                    }
                     // If we're going to delete this later, don't bother!
                     if road.internal_junction_road {
                         continue;
                     }
 
-                    if let Some(pl) = self.estimate_trimmed_geometry(road.id) {
-                        if road.src_i == i {
-                            if trim_roads_for_merging.contains_key(&(road.id, true)) {
-                                panic!(
```

Comment:
I got rid of these asserts because I hit a panic in `leeds_cycleway` and I couldn't justify why this is a problem. If we wind up setting this twice, it's fine? It'll be based on `road.center_line`, which goes through `update_geometry` like it should.